synapse/tools/cortex/layer/dump.py

```diff
@@ -0,0 +1,175 @@
+import os
+import tempfile
+import contextlib
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+import synapse.telepath as s_telepath
+
+import synapse.lib.cmd as s_cmd
+import synapse.lib.output as s_output
+import synapse.lib.msgpack as s_msgpack
+
+descr = '''
+Export node edits from a Synapse layer.
+'''
+
+# Helper to automatically close the tempfile and delete it on error
+@contextlib.contextmanager
+def tmpfile(dirn):
```

Comment:
Do we want to move this construct into `synapse.common` ? It seems useful.

Reply:
moved.